### PR TITLE
fix(connector): [coinbase][opennode][bitpay] handle error response

### PR DESCRIPTION
**Pull request by arvindpatel24**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Handle error response correctly for all crypto connectors.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [x] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/opennode/transformers.rs [hunk 1]
```diff
@@ -211,12 +211,9 @@ impl TryFrom<types::RefundsResponseRouterData<api::RSync, RefundResponse>>
 }
 
 //TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Deserialize)]
 pub struct OpennodeErrorResponse {
-    pub status_code: u16,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/opennode/transformers.rs:216]: Previously we had template code for ErrorResponse?

File: crates/router/src/connector/bitpay.rs [hunk 2]
```diff
@@ -109,9 +109,9 @@ impl ConnectorCommon for Bitpay {
 
         Ok(ErrorResponse {
             status_code: res.status_code,
-            code: response.code,
-            message: response.message,
-            reason: response.reason,
+            code: response.code.unwrap_or(format!("{}", res.status_code)),
```
Comment (jagan-jaya) [inline crates/router/src/connector/bitpay.rs:112]:
```suggestion
            code: response.code.unwrap_or_else(|| consts::NO_ERROR_CODE.to_string()),
```

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

File: crates/router/src/connector/bitpay.rs (hunk 2 above)
Comment (arvindpatel24) [reply crates/router/src/connector/bitpay.rs]: Done

File: crates/router/src/connector/opennode/transformers.rs (hunk 1 above)
Comment (arvindpatel24) [reply crates/router/src/connector/opennode/transformers.rs:216]: Hey yes it was there, but I was not aware of that to change.

**Review by jagan-jaya — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**